test/class_relationships_test.rb

```diff
@@ -1,77 +1,79 @@
-require 'class_relationships'
+require 'new_impl/class_relationships'
 require 'test_helper'
 
-[
-  [Object, Module, Class],
-  [ToyObject, ToyModule, ToyClass, 'toy_']
-].each do |_Object, _Module, _Class, prefix|
-  describe 'relationships between Object, Module and Class' do
-    include TestHelpers
+module NewImpl
+  [
+    [Object, Module, Class],
```

Comment:
Now that the test is inside a module (and, frankly, even if it weren’t, given the mindbending nature of this project) it might be both safer and clearer to fully-qualify these names (`::Object`, `::Module` and `::Class` respectively) to make it completely explicit that we’re referring to the constants defined inside the root namespace versus some other module.

Reply:
(Trivia: these constants actually live inside `Object`, so we can also refer to them as `::Object::Object`, `::Object::Module` and `::Object::Class` if we wish. 🤯)

Reply:
Oof, that's a bit mind-bendy 😂